src/api/app/models/review.rb

```diff
@@ -250,7 +250,7 @@ def matches_user?(user)
 
   def event_parameters(params = {})
     params = params.merge(_get_attributes)
-    params[:id] = id
+    params[:id] = bs_request.id
```

Comment:
You can't just change the event to suit your needs. You will have to do this traversal in `Notification`.

Also requests are identified by number.

Reply:
> You can't just change the event to suit your needs. You will have to do this traversal in `Notification`.

I think I'm not following

Reply:
one thing that henne said @danidoni  is that we don't use `bs_request.id` but `bs_request.number`

Reply:
Ummm good point, I forgot about it, thanks!

Reply:
> You can't just change the event to suit your needs. You will have to do this traversal in `Notification`.

Once I have a Notification, created from the event payload set here, I no longer have access to the request to fetch the id and fix it there.

And as we have set the `notifiable_type` to `BsRequest` [here](https://github.com/openSUSE/open-build-service/pull/9873/files#diff-ac49bda8ff4dfaeb507282adc2cea1f3R27), the current `notifiable_id` is the one for the `Review`, but the type is not, so the `notifiable` association is broken...

Those `event_parameters` are only used to build the event, so I think its safe to make the change here.

Reply:
> And as we have set the notifiable_type to BsRequest here

You don't *have* to do this right?

> Those event_parameters are only used to build the event, so I think its safe to make the change here.

The content of `Events` are send over the message bus. People might do something with it. If we *must* we can can change this, we never promised this to be stable. I also don't think you can actually do something with the review ID, but I haven't checked.

Reply:
I'll think about this and see if I can make a proposal with a better implementation